build.gradle

```diff
@@ -113,6 +122,15 @@ dependencies {
   smokeTestCompile sourceSets.main.runtimeClasspath
   smokeTestCompile group: 'org.springframework.boot', name: 'spring-boot-starter-test', version: versions.springBoot
   smokeTestCompile group: 'io.rest-assured', name: 'rest-assured', version: '3.0.7'
+
+  functionalTestCompile sourceSets.main.runtimeClasspath
+  functionalTestCompile(
```

Comment:
Some of these dependencies are probably not used. I'd remove those ones. Also, rest-assured could be in version 3.0.7, as for smoke tests.